### ISSUE TITLE
ArgumentException: Off-axis dimensions must match

### ISSUE THREAD

**Issue by RootbeerDev** [Unity-Technologies/barracuda-release#9]
Hello, I have a onnx model trained in pytorch. I can't share the model as the rights of it does not belong to me. It takes in an image as an input and outputs an image. I have tried various ways to initialize the tensor but none of it seems to work.

The image shape in the inspector is as such, image shape: (1,384,384,3)

```
        var model = ModelLoader.Load(modelSource);
        var worker = BarracudaWorkerFactory.CreateWorker(BarracudaWorkerFactory.Type.ComputePrecompiled, model);
        Dictionary<string, Tensor> inputs = new Dictionary<string, Tensor>();

        Tensor tensor = new Tensor();
        Texture2D texture = new Texture2D(384, 384, TextureFormat.RGBAFloat, false);
        tensor = new Tensor(texture);

        float[] floatarr = new float[384 * 384 * 3];

        Tensor tensor2 = new Tensor(1, 384, 384, 3, floatarr);
        inputs["image"] = tensor2;
        worker.Execute(inputs);

        var inputNames = model.inputs;   // query model inputs
        var outputNames = model.outputs; // query model outputs

```

> ArgumentException: Off-axis dimensions must match
> Barracuda.TensorExtensions.Concat (Barracuda.TensorShape[] shapes, System.Int32 axis) (at <a57b068757aa4cec9131100114e8a39b>:0)
> Barracuda.ModelAnalyzer.ListTemporaryTensorShapes (Barracuda.Model model, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes, System.Collections.Generic.IDictionary`2[System.String,Barracuda.TensorShape]& shapesByName) (at <a57b068757aa4cec9131100114e8a39b>:0)
> Barracuda.ModelAnalyzer.ListTemporaryTensorShapes (Barracuda.Model model, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
> Barracuda.ModelAnalyzer.FindLargestNecessaryTensorShape (Barracuda.Model model, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
> Barracuda.GenericVarsWithPreallocation.PrepareStorage (Barracuda.Model model, Barracuda.IOps ops, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
> Barracuda.GenericWorker.PrepareForInput (System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
> Barracuda.GenericWorker.AddInput (System.String name, Barracuda.Tensor x) (at <a57b068757aa4cec9131100114e8a39b>:0)
> Barracuda.GenericWorker.Execute (System.Collections.Generic.IDictionary`2[TKey,TValue] inputs) (at <a57b068757aa4cec9131100114e8a39b>:0)
> BarrcudaScript.Start () (at Assets/BarrcudaScript.cs:26)

Any assistance would be greatly appreciated. Thank you.

**Comment by kdsh7** [comment]
I'm getting the exact same issue with a model trained on tensorflow with YOLOv2. The shape seems fine:
 (1, 416, 416, 3)

```        var inputs = new Dictionary<string, Tensor>();
        input = new Tensor(inferenceTexture, 3);
        input.Print();
        inputs.Add("input", input);

        engine.Execute(inputs); 


And the error:

```ArgumentException: Off-axis dimensions must match
Barracuda.TensorExtensions.Concat (Barracuda.TensorShape[] shapes, System.Int32 axis) (at <a57b068757aa4cec9131100114e8a39b>:0)
Barracuda.ModelAnalyzer.ListTemporaryTensorShapes (Barracuda.Model model, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes, System.Collections.Generic.IDictionary`2[System.String,Barracuda.TensorShape]& shapesByName) (at <a57b068757aa4cec9131100114e8a39b>:0)
Barracuda.ModelAnalyzer.ListTemporaryTensorShapes (Barracuda.Model model, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
Barracuda.ModelAnalyzer.FindLargestNecessaryTensorShape (Barracuda.Model model, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
Barracuda.GenericVarsWithPreallocation.PrepareStorage (Barracuda.Model model, Barracuda.IOps ops, System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
Barracuda.GenericWorker.PrepareForInput (System.Collections.Generic.IDictionary`2[TKey,TValue] inputShapes) (at <a57b068757aa4cec9131100114e8a39b>:0)
Barracuda.GenericWorker.AddInput (System.String name, Barracuda.Tensor x) (at <a57b068757aa4cec9131100114e8a39b>:0)
Barracuda.GenericWorker.Execute (System.Collections.Generic.IDictionary`2[TKey,TValue] inputs) (at <a57b068757aa4cec9131100114e8a39b>:0)```

**Comment by mantasp** [comment]
@Widgertius @kdsh7 could you please post links to the model you are trying to run? (no need for proper weights it might be initialized with random)

**Comment by robinydv** [comment]
@kdsh7 I am trying to use YOLOv3 from TF model in unity. Does the v2 model have a FusedBatchNormv3 layer? I believe that the errors I am getting relate to that.

**Comment by RootbeerDev** [comment]
@mantasp  Hi sorry for the late reply,

Please download the models with randomized weights here
https://mut1nyjd.stackstorage.com/s/y0lOWiVYczNnhvQ

**Comment by RootbeerDev** [comment]
So upon trying the new 0.50 build, there are different errors

This are the values when I click on the model in the inspector.

![OnInpsectorGUI](https://user-images.githubusercontent.com/10050395/74217947-def35e80-4ce3-11ea-9e7b-76e61f292e2c.PNG)

which causes a bunch of null reference exceptions in the file ONNXModelImporterEditor.cs, function ListUIHelper, Debug.Assert(descriptions.Count == n) in line 126;

And then in TensorExtensions.cs, in function ApplyStridedSlice, the assert  Assert.IsTrue(starts[i] < counts[i]); is thrown on line 474 when I actually try running the model.

**Comment by mantasp** [comment]
Thank you for additional details! 
StridedSlice issue should be fixed in the next release. Will try your models on it to see if any other issues exist.

**Comment by kdsh7** [comment]
Just to add to this (sorry for the late replies everyone) - the converter scripts in 0.60 seem to have done the job. This now works in editor.

However on Android (Samsung Galaxy S10) it's throwing GLSL link errors. I'll raise another issue so as to not muddy up this support thread.
